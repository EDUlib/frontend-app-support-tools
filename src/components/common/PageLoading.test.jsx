import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import PageLoading from './PageLoading';

describe('<PageLoading />', () => {
  it('does not render without message', () => {
    const wrapper = shallow(<PageLoading srMessage="" />);

    expect(wrapper.find('.sr-only')).toHaveLength(0);
  });
  it('renders expected message', () => {
    const message = 'Loading...';

    const wrapper = shallow(<PageLoading srMessage={message} />);
    const srElement = wrapper.find('.sr-only');

    expect(srElement).toHaveLength(1);
    expect(srElement.text()).toEqual(message);
  });
  it('snapshot matches correctly', () => {
    const tree = renderer.create(<PageLoading srMessage="Loading" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
