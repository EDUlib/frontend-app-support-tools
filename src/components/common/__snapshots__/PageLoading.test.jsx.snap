// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLoading /> snapshot matches correctly 1`] = `
<div>
  <div
    className="d-flex justify-content-center align-items-center flex-column"
    style={
      Object {
        "height": "50vh",
      }
    }
  >
    <div
      className="spinner-border text-primary"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading
      </span>
    </div>
  </div>
</div>
`;
